docs/Developers/Debug_Bears.rst

```diff
@@ -0,0 +1,77 @@
+Debug Bears
+===========
+
+This document provide a brief overview of coala's Debugging Interface.
+The Debugging Interface will help users to debug Bear code and step through
+it using pdb interface
+
+Usage of coala's Debugger
+=========================
+
+Command Line Interface
+----------------------
+
+Users can specify the bear they want to debug using ``--debug-bears``, i.e.
+
+.. code:: shell
+
+    $coala -b PEP8Bear,HelloWorldBear -f <filename> --debug-bears HelloWorldBear
+
+Below is the code given for a simple ``HelloWorldBear`` that prints a debug
+message for each file:
+
+.. code:: python
+
+    import logging
+
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class HelloWorldBear(LocalBear):
+        def run(self,
+                filename,
+                file):
+            logging.info('Please follow newcomer guide.')
+            logging.debug('Hello World! Checking file {}.'.format(filename))
+
+After invoking the coala's debugger it will step into the ``run()`` method of a
+HelloWorldBear and help users to debug Bear code with passed settings.
+
+If bear names are not specified for ``--debug-bears`` then it will by
+default debug all the bears passed through ``--bears`` or ``-b`` argument.
+
+.. code:: shell
+
+    $ coala --bears HelloWorldBear -files <filename> --debug-bears
+
+.. note::
+
+    A bear may depend on results from different bears. The debugger will debug
+    all bears on which a bear is dependent on as well.
+
+coafile
+-------
+
+Users can debug bears using ``.coafile``:
+
+::
+
+    [all]
+    bears = PEP8Bear,MypyBear
+    files = <filename>
+    debug_bears = PEP8Bear
+
+or
+
+::
+
+    [all]
+    bears = PEP8Bear,MypyBear
+    files = <filename>
+    debug_bears = True
+
+.. note::
+
+    Pdb's quit-command (``q``) has been remapped so coala continues its
+    normal execution without aborting. So, ``quit`` or `q` will first clear
+    all the breakspoints then continue ( ``c`` or ``continue`` ) execution.
```

Comment:
This should go into the above proposed "Details" paragraph, as it's not directly related to the invocation of the debugger within a `coafile`;)